src/common/wallet-state.ts

```diff
@@ -433,66 +409,8 @@ function useWalletState(initialState?: Partial<WalletStateParams>): WalletData {
     setTotalBalance(balance)
   }
 
-  const loadTransactionHistory = async (): Promise<void> => {
-    const currentAddress = getCurrentAddress()
-    const currentBlock = await web3.eth.getBlockNumber()
-
-    const web3transactions: readonly AccountTransaction[] = await web3.mantis
-      .getAccountTransactions(currentAddress, Math.max(0, currentBlock - 999), currentBlock)
-      .then((txns) => {
-        //TEMPORARY: Deduplicate pending transactions if they are also present in chain
-        //Once ETCM-363 is there this code will be in better place
-        const confirmedHashes = pipe(
-          txns,
-          readonlyArray.filterMap((tx) =>
-            tx.blockNumber != null ? option.some(tx.hash) : option.none,
-          ),
-          (hashes) => new Set(hashes),
-        )
-
-        return pipe(
-          txns,
-          readonlyArray.filterMap((tx) =>
-            tx.isPending && confirmedHashes.has(tx.hash) ? option.none : option.some(tx),
-          ),
-        )
-      })
-
-    const newTransactions = await Promise.all(
-      web3transactions.map(
-        async (tx): Promise<Transaction> => {
-          return {
-            ...tx,
-            timestamp: tx.timestamp || null,
-            gasUsed: tx.gasUsed || null,
-            value: asWei(tx.value),
-            gasPrice: asWei(tx.gasPrice),
-            fee: tx.isOutgoing
-              ? asWei(new BigNumber(tx.gasPrice).times(tx.gasUsed || tx.gas))
-              : asWei(0),
-            direction: tx.isOutgoing ? 'outgoing' : 'incoming',
-            status: getStatus(tx, currentBlock),
-            contractAddress:
-              tx.to == null
-                ? (await web3.eth.getTransactionReceipt(tx.hash))?.contractAddress || null
-                : null,
-          }
-        },
-      ),
-    ).then((txns) =>
-      pipe(
-        transactionsOption,
-        option.getOrElseW(() => []),
-        mergeTransactions(txns),
-      ),
-    )
-
-    setTransactions(some(newTransactions))
-    await loadBalance(newTransactions)
-  }
-
   const load = (
-    loadFns: Array<() => Promise<void>> = [loadTransactionHistory, loadAccounts],
+    loadFns: Array<() => Promise<void>> = [() => loadBalance(transactions), loadAccounts],
```

Comment:
I don't really like this default here, I would rather just call the function with those functions as parameter. It seems to be quite generic function not sure why should it load balance and accounts specifically. Moreover...don't we want to handle loading balance separately from loading transactions?

Reply:
But that is what is happening (more or less), I just removed the argument from there since there's literally single call of it.

That's another thing to be improved, but I didn't really want to even think about improving these things - that took me enough time and the PR is big enough on its own to include some additional improvements.